flake-info/src/commands/flake_info.nix

```diff
@@ -82,20 +82,22 @@ let
 
         # Replace functions by the string <function>
         substFunction = x:
-          if x ? _type && (x._type == "literalExample" || x._type == "literalExpression" || x._type == "literalDocBook") then
-            x.text
+          if x ? _type && ( x._type == "literalExample" || x._type == "literalExpression" || x._type == "literalDocBook") && builtins.isString x.text then
+            { __type = x._type; __value =  x.text; }
+          else if x ? type && x.type == "derivation" then 
+            { __type="derivation"; __value = x; }
           else if builtins.isAttrs x then
-            lib.mapAttrs (name: substFunction) x
+            {__type = "set"; __value = lib.mapAttrs (_:substFunction )  x; }
           else if builtins.isList x then
-            map substFunction x
+            { __type = "list"; __value = map substFunction x; }
           else if lib.isFunction x then
-            "<function>"
+            { __type = "function"; }
           else
-            x;
+            { __type = "value" ; __value = x; };
```

Comment:
Do we really need all this? My idea was to just have `substFunction` do what it's supposed to do (i.e. replace functions with `<function>`), and do the handling of `_type`d attrsets in Rust.

Reply:
either way works i guess, but you are right its probably more performant and easier to do in rust.
However that is not the main problem here.
Its more the fact that "literalExpressions" while supposed to be strings (of a string - the literal value) turn out to be all kinds of stuff and therefore make it hard to reserialize them nicely